### DISCUSSION TITLE
[WASM] AOT build fails. unoapp-winui-net6 template

### DISCUSSION THREAD

**Discussion by emceelovin** [unoplatform/uno#7864] [General]
I've tried everything in issue #7740, but I can't seem to get the AOT release to build. The supplied screenshot of the build log is as far as can dig into the error. I've troubleshooted and built the project with all of the steps in [Troubleshooting build issues](https://platform.uno/docs/articles/uno-builds-troubleshooting.html). I followed all of the steps in the AOT [Build .NET AOT for WebAssembly in Visual Studio with Uno Platform](https://platform.uno/blog/build-net-aot-for-webassembly-in-visual-studio-with-uno-platform/). However, while I did all of those steps, a window did not popup asking me to run a script, but I manually ran everything in the supplied script in the article. Except, I got some errors. I had to install dotnet-3.1. dotnet-3.0 wouldn't install. It required some kind of library dependency. When I build my wasm project in release and AOT enabled, the build partially succeeds. The console pops up, and then I get this build error:

I have everything set to diagnostics level output, but I cant seem to get more information out of this error. I've built with the most verbose settings and all of the commands listed in the troubleshooting. Still can't get this to build.

[msbuid.binlog.zip](https://github.com/unoplatform/uno/files/7900317/msbuid.binlog.zip)

![image](https://user-images.githubusercontent.com/55720292/150207288-107efa46-5c02-4e18-9d21-f743be129f53.png)

*EDIT*
Apparently, Uno.Wasm.Bootstrap.Cli got uninstalled somehow, but once I installed it, I get these errors in addition to the other errors no
[binlog.zip](https://github.com/unoplatform/uno/files/7900399/binlog.zip)
w

![image](https://user-images.githubusercontent.com/55720292/150209509-a97094d0-81fa-4ac6-8a47-14dc75461f67.png)

This is the latest error I get with these as my project dependencies. If I uncommenting the MVC dependencies don't fix anything:
![image](https://user-images.githubusercontent.com/55720292/150217246-0ebbfb4b-d060-4911-b49e-0a4275831372.png)

**Comment by jeromelaban** [comment]
Thanks for the report. You seem to be having an extra option that's not needed anymore `<ExtraEmccFlags Include="-lidbfs.js" />`. 

Can you try removing it and see if it helps ?

**Reply by emceelovin** [reply]
Oh ok. It was on the build AOT docs, so I didn't know it wasnt needed. Give me a second.

**Reply by jeromelaban** [reply]
Interesting, can you point me to the documentation that suggested this option?

**Reply by emceelovin** [reply]
it's the Build .NET AOT link in the OP

**Reply by emceelovin** [reply]
It might be a second. I updated to VS2022 17.1 preview 3, and I knew i shouldn't have. Getting a lot of errors now that aren't even errors. Plus some new errors.

**Reply by emceelovin** [reply]
I should have stayed on preview 2. UWP and everything else works, but WASM is a huge mess with preview 3.

**Reply by emceelovin** [reply]
Things kind of hit the fan
![image](https://user-images.githubusercontent.com/55720292/150256399-0a9aa546-aefb-45e6-9051-ae5f7664c14d.png)

**Reply by jeromelaban** [reply]
Ah I see. It's an older article that we have not updated and we should link to a more recent version of it. Up to date detailed information about AOT [is available here](https://github.com/unoplatform/Uno.Wasm.Bootstrap/#runtime-execution-modes).

**Reply by emceelovin** [reply]
Thank you! When I get home, I'm gonna have to get back on visual studio preview 2 so I can see if that fixes it. Thanks for the help.

**Reply by jeromelaban** [reply]
We have not tested preview 3 yet, but yes, it's likely that there are changes :) Thanks for reporting.

**Reply by emceelovin** [reply]
No problem! Really appreciate all that's being done, and I like to help anytime I can when I find an issue.

**Reply by emceelovin** [reply]
Well now that I am able to build Uno projects again on the latest 17.1, I still am not able to get wasm AOT working. This is the error i get, and the package references my project has:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <NoWarn>NU1701</NoWarn>
	  <WasmShellMonoRuntimeExecutionMode Condition="'$(Configuration)'=='Release'">InterpreterAndAOT</WasmShellMonoRuntimeExecutionMode>
  </PropertyGroup>
	<PropertyGroup Condition="'$(Configuration)'=='Debug'">
    <MonoRuntimeDebuggerEnabled>true</MonoRuntimeDebuggerEnabled>
    <DefineConstants>$(DefineConstants);TRACE;DEBUG</DefineConstants>
    <DebugType>portable</DebugType>
    <DebugSymbols>true</DebugSymbols>
    <!--
		IL Linking is disabled in Debug configuration.
		When building in Release, see https://platform.uno/docs/articles/features/using-il-linker-webassembly.html
		-->
    <WasmShellILLinkerEnabled>false</WasmShellILLinkerEnabled>
  </PropertyGroup>
  <ItemGroup>
    <Content Include="Assets\SplashScreen.png" />
  </ItemGroup>
  <ItemGroup>
    <UpToDateCheckInput Include="..\OneGreatTeam.Shared\**\*.xaml" />
  </ItemGroup>
  <ItemGroup>
    <EmbeddedResource Include="WasmCSS\Fonts.css" />
    <EmbeddedResource Include="WasmScripts\AppManifest.js" />
  </ItemGroup>
  <ItemGroup>
    <LinkerDescriptor Include="LinkerConfig.xml" />
  </ItemGroup>
  <ItemGroup>
    <!--
		This item group is required by the project template because of the
		new SDK-Style project, otherwise some files are not added automatically.

		You can safely remove this ItemGroup completely.
		-->
    <None Include="Program.cs" />
    <None Include="LinkerConfig.xml" />
    <None Include="wwwroot\web.config" />
</ItemGroup>
<ItemGroup>
    <PackageReference Include="Microsoft.Windows.Compatibility" Version="5.0.0" />
    <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
    <PackageReference Include="System.Reflection.Emit.Lightweight" Version="4.7.0" />
    <PackageReference Include="System.Threading.Channels" Version="6.0.2-mauipre.1.22054.8" />
    <PackageReference Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.0.1" />
    <PackageReference Include="Uno.WinUI.WebAssembly" Version="4.1.0-dev.545" />
    <PackageReference Include="Uno.WinUI.RemoteControl" Version="4.0.11" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.0.11" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="3.2.0-dev.24" />
    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="3.2.0-dev.24" />
    <PackageReference Include="Uno.Wasm.Bootstrap.Cli" Version="1.1.0-dev.445" />
</ItemGroup>
<Import Project="..\OneGreatTeam.Shared\OneGreatTeam.Shared.projitems" Label="Shared" Condition="Exists('..\OneGreatTeam.Shared\OneGreatTeam.Shared.projitems')" />
</Project>
```

even if i add the Microsoft.AspnetCore.Mvc.Core project to my package references, I still can't get it to build. If i remove the Uno.Wasm.Bootstrap.Cli package reference, the build process just sits at loading the <project>.wasm.dll. When I look at task manager, a bunch of clang+++ and python.exe's are doing something though. They come and then disappear. Come and disappear. If i drop the Uno.Wasm.Bootstrap.Cli package to 1.1.0-dev.416, I no longer get the MVC error, but it still just sits at "1>OneGreatTeam.Wasm -> C:\Users\jeffc\source\repos\One Great Team\uno\OneGreatTeam\OneGreatTeam.Wasm\bin\Release\net6.0\OneGreatTeam.Wasm.dll"

![image](https://user-images.githubusercontent.com/55720292/153742852-54aa8bf5-42f3-4281-878e-0afc7a5ae664.png)

**Reply by jeromelaban** [reply]
Thanks for the update. The `Uno.Wasm.Bootstrap.Cli` package is deprecated and is not needed anymore.

When troubleshooting issues, make sure to start from the official templates to check if a vanilla configuration works, either by creating a [new project from visual studio](https://platform.uno/docs/articles/get-started-vs.html), or by using [`dotnet new` templates](https://platform.uno/docs/articles/get-started-dotnet-new.html).

**Reply by emceelovin** [reply]
Oh ok. I didn't know it was deprecated. There are quite a few wasm AOT docs, but I think i did notice that the github wasm AOT doc didn't reference that package. I can't remember if i created this project via dotnet or VS. I'm pretty sure I created it via dotnet new uno-app-winui-net6. I will try creating a new project and see if it works, and if it does, I will just move everything over, but I figured I would give any info I could to help things along. Thank you for the help once again. It probably is an older template.

**Reply by emceelovin** [reply]
I just created a new unoapp-winui-net6 template, which had a template update even though i checked the template version the other day and was on the latest, and the AOT wasm built, so thank you!

EDIT: Apparently, Interpreter likes to work, but InterpreterAndAOT just sits at loading the .wasm.dll and pinning my CPU at 100%. I guess its trying to compile something, but a test project is tiny. It shouldn't take this long?

I'm back to WindowsAppSDK version errors again, and I forgot how i cleared those up last time. Gonna try and get this together....
